src/modules/cpu/kernel/image_sum.hpp

```diff
@@ -0,0 +1,860 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus image_sum_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                      RpptDescPtr srcDescPtr,
+                                      Rpp32f *imageSumArr,
+                                      RpptROIPtr roiTensorPtrSrc,
+                                      RpptRoiType roiType,
+                                      RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(srcDescPtr->n)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp8u *srcPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+
+        Rpp32u alignedLength = (bufferLength / 24) * 24;
+        Rpp32u vectorIncrement = 24;
+        Rpp32u vectorIncrementPerChannel = 8;
+
+        // Image Sum without fused output-layout toggle (NCHW)
+        if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            alignedLength = (bufferLength / 8) * 8;
+            vectorIncrement = 8;
+            Rpp64f sum = 0.0;
+            Rpp64f sumAvx[4] = {0.0};
+            
+            for(int c = 0; c < layoutParams.channelParam; c++)
+            {
+                Rpp8u *srcPtrRow;
+                srcPtrRow = srcPtrChannel;
+#if __AVX2__
+                __m256d psum = _mm256_setzero_pd();
+#endif
+                for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+                {
+                    Rpp8u *srcPtrTemp;
+                    srcPtrTemp = srcPtrRow;
+
+                    int vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                    {
+#if __AVX2__
+                        __m256d p1[2];
+                        rpp_simd_load(rpp_load8_u8_to_f64_avx, srcPtrTemp, p1);
```

Comment:
For sum computation, I understand f16/f32 may need to be processed in f32/d64.
But can't we use _mm256_add_epi32() or _mm256_add_epi64() instead of _mm256_add_pd() for U8 or I8 since they are all integers to be added? to save on cycles?

Reply:
Sure, yeah will replace _mm256_add_pd() with  _mm256_add_epi64() for U8/I8.

Reply:
32 bit can store upto a range of 4,294,967,295 and when we have our input 4k image with all pixels set to 255 then this would mean summation result of 2,25,60,76,800 and if we move to an 8k image then we might need 64 bit and so far as we use f32 for internal computes given any datatype we need to convert u8 to f32 and using _mm256_add_epi64 there with float values gives undefined results.

I also thought about using u8 directly for this instead of using f32 for internal computes but this might lead to mismatch with HIP(as we use floating point computes there) when we check for golden outputs.

I also wanted to check on how to create golden outputs for these as summations computed via HIP might not be very accurate due to thread parallelization.